app/src/main/java/com/yacgroup/yacguide/activity_properties/RockSearchable.kt

```diff
@@ -21,46 +21,64 @@ import android.content.Intent
 import android.view.ViewGroup
 import android.widget.*
 import androidx.appcompat.app.AppCompatDialog
+import androidx.viewbinding.ViewBinding
 import com.yacgroup.yacguide.R
 import com.yacgroup.yacguide.RockActivity
 import com.yacgroup.yacguide.TableActivityWithOptionsMenu
 import com.yacgroup.yacguide.database.comment.RockComment
+import com.yacgroup.yacguide.databinding.RockSearchDialogBinding
 import com.yacgroup.yacguide.utils.FilterSpinner
 import com.yacgroup.yacguide.utils.IntentConstants
 import com.yacgroup.yacguide.utils.FilterSpinnerListener
 
-class RockSearchable(private val _activity: TableActivityWithOptionsMenu) : ActivityProperty, FilterSpinnerListener {
+class RockSearchable<ViewBindingType: ViewBinding>(private val _activity: TableActivityWithOptionsMenu<ViewBindingType>) : ActivityProperty, FilterSpinnerListener {
 
     private var _maxRelevanceId: Int = RockComment.NO_INFO_ID
 
     override fun getMenuGroupId() = R.id.group_rock_search
 
     override fun onMenuAction(menuItemId: Int) {
-        val searchDialog = AppCompatDialog(_activity, R.style.AppTheme_Dialog)
-        searchDialog.setContentView(R.layout.rock_search_dialog)
-        searchDialog.window?.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
+        val searchDialog = AppCompatDialog(_activity, R.style.AppTheme_Dialog).apply {
```

Comment:
Have we forgotten to use the DialogWidgetBuilder here or is there a reason why still using `AppDompatDialog`?